prow/plugins/plugins.go

```diff
@@ -369,6 +369,14 @@ type Lgtm struct {
 	// StoreTreeHash indicates if tree_hash should be stored inside a comment to detect
 	// squashed commits before removing lgtm labels
 	StoreTreeHash bool `json:"store_tree_hash,omitempty"`
+	// WARNING: Read below for potential risk before you enable this option.
+	// StickyForCollaborators indicates if LGTM is sticky for PRs authored by collaborators.
+	// This means that collaborators are trusted to not introduce bad code after the initial
+	// LGTM, and it eliminates the need to re-lgtm minor fixes/updates. If the PR is authored
+	// by non-collaborator, this option does not apply.
+	// This disables the security mechanism that prevents a malicious collaborator (or
+	// compromised collaborator GitHub account) from merging arbitrary code.
+	StickyForCollaborators bool `json:"sticky_for_collaborators,omitempty"`
```

Comment:
By default prow will trust repo collaborators, but there is also a flag to only trust org members.

We should consider following this same value here. Aka if we only trust org members, we should only have a sticky lgtm for org members:

https://github.com/kubernetes/test-infra/blob/2264543fd4ad973cfb838a9b4a17cf37dea81f76/prow/plugins/plugins.go#L395-L397

Reply:
> By default prow will trust repo collaborators, but there is also a flag to only trust org members.
>
> We should consider following this same value here. Aka if we only trust org members, we should only have a sticky lgtm for org members:

@fejta are you suggesting LGTM should be sticky for all collaborators by default, and the option controls if the sticky LGTM applies to org members?

That seems to go against the desire to make sure all changes to the last bit are reviewed by default. Please advice.

Reply:
And perhaps I misunderstood. 

Looking at https://github.com/kubernetes/test-infra/blob/2264543fd4ad973cfb838a9b4a17cf37dea81f76/prow/plugins/trigger/trigger.go#L142, perhaps you mean this should be an option for "trusted users", which is derived from trigger.trustedUser().

That way, the OnlyOrgMembers config can kick in as well.

Does it sound right?

Reply:
Yes, I want `/ok-to-test` and sticky LGTM to trust the same set of users.

Reply:
I think Erick is right, but that change seems unrelated to this PR. The `lgtm` plugin already checks if users are collaborators and doesn't consider the value of `OnlyOrgMembers`. This PR doesn't change that.
If we want to change the behavior of `lgtm` and share config with `trigger`, we should move that config out of the `trigger` config section and add announcements about the breaking changes. I think that would be easier to do in a separate PR.
WDYT?

Reply:
And to clarify:

Do you suggest (in the order of perceived difficulty):
1. Add another OnlyOrgMembers boolean option in Lgtm to control this behavior in Lgtm?
2. Use Trigger.OnlyOrgMembers to control both Trigger and Lgtm plugins?
3. Refactor Trigger.OnlyOrgMembers to a common place and make both plugins use it?

Reply:
I agree that whether or not we trust the reviewer enough to add an LGTM label is unrelated to this PR.

However, whether or not we trust the PR author enough to keep or remove the LGTM label when a new commit is pushed by the PR author is different. This is the same decision as whether or not we trust the PR author enough to auto-trigger testing without an `/ok-to-test` comment. It needs to follow the same logic (we may or may not trust a second org. We may or may not trust non-repo collaborators).